crates/supervisor/core/src/supervisor.rs

```diff
@@ -78,6 +78,9 @@ pub trait SupervisorService: Debug + Send + Sync {
     /// Returns list of supervised [`ChainId`]s.
     fn chain_ids(&self) -> impl Iterator<Item = ChainId>;
 
+    /// Returns mapping of supervised [`ChainId`]s to their [`ChainDependency`] config.
+    fn dependency_set(&self) -> DependencySet;
```

Comment:
```suggestion
    fn dependency_set(&self) -> &DependencySet;
```
usually makes more sense to return a reference and let the caller handle the clone if it's requited in caller's context